src/Connection.js

```diff
@@ -30,7 +48,12 @@ module.exports = class Connection extends events.EventEmitter {
     }
 
     send(msg) {
-        const serialized = msg.serialize()
+        let serialized
+        if (this.preferredPayloadVersion) {
+            serialized = msg.serialize(0, this.preferredPayloadVersion)
```

Comment:
Probably unnecessary `if-else`: if you call `msg.serialize(0, this.preferredPayloadVersion)` and `this.preferredPayloadVersion` is `undefined`, you'll get the same thing as `msg.serialize()`.

What we could do actually is add support for both a `protocolVersion` and `payloadVersion` query params (instead of just `version`, which is ambiguous), then always call `msg.serialize(this.protocolVersion, this.payloadVersion)`.

Then in `serialize(...)`, `protocolVersion` defaults to `0` and `payloadVersion` to `28` (which they already do).